### PR TITLE
v.parser: fix -translated for modules with C interop

### PR DESCRIPTION
**Pull request by div72**
Fixes half of #11365, `v -translated self` is still broken.

### CODE REVIEW

**Comment by JalonSolov** [comment]
Why would you want `v -translated self` to work at all?  `-translated` is for use with code that `c2v` generated, for example, and V is not that.

**Comment by div72** [comment]
@JalonSolov @vincenzopalazzo -translated should not break V code, it should just relax some requirements. The point is not the V compiler, but that ALL valid V programs should be compiled without errors. Transpiled programs may depend on some helper V code.

The V compiler is just the best test for this.

**Comment by JalonSolov** [comment]
I agree that all V programs should compile.  I just think you don't need `-translated` on a program that wasn't translated.  There should be a message that says it wasn't needed, and it should be ignored in that case.

Yes, I know... how do you know if something was translated or not?  Well, at the very least, you could do this when `self` is what is being compiled.

**Comment by spytheman** [comment]
I agree with @div72 - V programs should be compilable in -translated mode. 
I think that modules containing mixed code - both files produced by `c2v` *and* normal .v files, written by a V programmer, will be common, especially for projects, that are translated piece by piece.

**Comment by spytheman** [comment]
An alternative would be specifying the mode inside each .v file itself, like we have a `[has_globals]` module tag 🤔 .

**Comment by JalonSolov** [comment]
My point is that `-translated` "relaxes the rules", and that should _only_ be allowed on translated files, not V code in general.  Otherwise, why not just make `-translated` the default?

**Comment by spytheman** [comment]
A module can have multiple .v files. One or more of them, can be the result of c2v, and the rest of them can be manually written by V programmers (imagine a codebase that was initially translated, so all the files were the result of `c2v`, but was later extended with pure V code). 

Currently `-translated` is a switch, that should be given at the command line, and affects *all .v files*, including the pure V ones => turning it on is an all, or nothing thing. 

The stricter mode (the default) is valuable, since it enables more checks and thus can catch more errors earlier, and so it is better to use it for as much files as possible.

TLDR: I think that the granularity of `-translate` is too coarse.

**Comment by div72** [comment]
@spytheman I really like the idea of a module tag. It would also make it trivial to add tests for -translated regressions like this. Generating the tag is also easier than hoping someone would read the docs of the transpiler to run the file with -translated.

**Comment by JalonSolov** [comment]
It only leaves the issue of when/how to remove said module tag... it would be up to a dev to do it manually, and you _know_ some won't do it, but... a problem for another day, I suppose.
